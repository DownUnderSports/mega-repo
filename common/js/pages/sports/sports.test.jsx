import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router, Route } from 'react-router-dom';

import SportsPage from './sports';

describe('Pages - SportsPage', () => {
  const div = document.createElement('div');

  const createSportsPage = ({...props}) => {
    ReactDOM.render((
      <Router>
        <Route component={(routeProps) => <SportsPage {...props} {...routeProps} path="/sports" />}/>
      </Router>
    ), div);
    return div.querySelector('section')
  }

  describe('Index', () => {
    it('renders by default')
    it('has a sport lookup form')

    it('is snapshotable', () => {
      const tree = renderer
        .create(
          <Router initialEntries={['/sports']}>
            <Route component={SportsPage} path="/sports" />
          </Router>
        )
        .toJSON();
      expect(tree).toMatchSnapshot()
    })
  })

  describe('Show', () => {
    it('pulls up data on a sport')
    it('data can be edited by double clicking')

    it('is snapshotable', () => {
      const tree = renderer
        .create(
          <Router initialEntries={['/sports/1']}>
            <Route component={SportsPage} path="/sports" />
          </Router>
        )
        .toJSON();
      expect(tree).toMatchSnapshot()
    })
  })

  describe('New', () => {
    it('renders the sport input form')

    it('is snapshotable', () => {
      const tree = renderer
        .create(
          <Router initialEntries={['/sports/new']}>
            <Route component={SportsPage} path="/sports" />
          </Router>
        )
        .toJSON();
      expect(tree).toMatchSnapshot()
    })
  })
})
