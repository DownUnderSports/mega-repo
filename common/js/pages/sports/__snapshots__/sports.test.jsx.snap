// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages - SportsPage Index is snapshotable 1`] = `
<div
  style={
    Object {
      "display": "block",
      "width": "100%",
    }
  }
>
  <h1
    className="text-center"
  >
    SUBMITTING...
  </h1>
  <div
    className="la-ball-atom la-vw-sm"
    style={
      Object {
        "color": "#00F",
        "marginLeft": "auto",
        "marginRight": "auto",
        "zIndex": 0,
      }
    }
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;

exports[`Pages - SportsPage New is snapshotable 1`] = `
<div
  style={
    Object {
      "display": "block",
      "width": "100%",
    }
  }
>
  <h1
    className="text-center"
  >
    SUBMITTING...
  </h1>
  <div
    className="la-ball-atom la-vw-sm"
    style={
      Object {
        "color": "#00F",
        "marginLeft": "auto",
        "marginRight": "auto",
        "zIndex": 0,
      }
    }
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;

exports[`Pages - SportsPage Show is snapshotable 1`] = `
<div
  style={
    Object {
      "display": "block",
      "width": "100%",
    }
  }
>
  <h1
    className="text-center"
  >
    SUBMITTING...
  </h1>
  <div
    className="la-ball-atom la-vw-sm"
    style={
      Object {
        "color": "#00F",
        "marginLeft": "auto",
        "marginRight": "auto",
        "zIndex": 0,
      }
    }
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
